cirq-google/cirq_google/ops/common_gate_families.py

```diff
@@ -0,0 +1,362 @@
+
+    1. `cirq.FSimGate`, `cirq.PhasedFSimGate`
+    2. `cirq.ISwapPowGate`, `cirq.PhasedISwapPowGate`
+    3. `cirq.CZPowGate`
+    4. `cirq.IdentityGate`
+
+    The gate family also allows an option to accept parameterized gates, assuming that the correct
+    parameters would eventually be filled.
+
+    For example,
+
+    1. To convert to/from any of the non-parameterized `POSSIBLE_FSIM_GATES` types:
+    >>> gf = cirq_google.FSimGateFamily()
+    >>> assert gf.convert(cirq.FSimGate(np.pi/4, 0), cirq.ISwapPowGate) == cirq.SQRT_ISWAP_INV
+    >>> assert gf.convert(cirq.PhasedFSimGate(0, 0, 0, 0, np.pi), cirq.CZPowGate) == cirq.CZ
+    >>> assert gf.convert(cirq.FSimGate(-np.pi/2, sympy.Symbol("t")), cirq.ISwapPowGate) is None
+
+    2. To convert to/from any of the parameterized `POSSIBLE_FSIM_GATES` types (assuming correct
+        value of the parameter would be filled in during paramter resolution):
+    >>> gf = cirq_google.FSimGateFamily(allow_symbols = True)
+    >>> theta, phi = sympy.Symbol("theta"), sympy.Symbol("phi")
+    >>> assert gf.convert(cirq.FSimGate(-np.pi/4, phi), cirq.ISwapPowGate) == cirq.SQRT_ISWAP
+    >>> assert gf.convert(cirq.PhasedFSimGate(theta, 0, 0, 0, np.pi), cirq.CZPowGate) == cirq.CZ
+    >>> assert gf.convert(cirq.FSimGate(theta, phi), cirq.IdentityGate) == cirq.IdentityGate(2)
+
+    3. To accept instances of `gate_types_to_check` based on type / value equality check against
+    gates in `gates_to_accept`, possibly accepting parameterized instances (assuming correct
+    parameter value would be filled in during parameter resolution) based on `allow_symbols`:
+    >>> gf = cirq_google.FSimGateFamily(
+    >>>     gates_to_accept=[cirq.SQRT_ISWAP, cirq.CZPowGate, cirq_google.SYC],
+    >>>     gate_types_to_check=[cirq.FSimGate],
+    >>>     allow_symbols=True,
+    >>> )
+    >>> theta, phi = sympy.Symbol("theta"), sympy.Symbol("phi")
+    >>> assert cirq.FSimGate(theta, phi) in gf # Assumes correct theta/phi will be substituted.
+    >>> assert cirq_google.SYC in gf # SYC
+    >>> assert cirq.FSimGate(0, np.pi / 2) in gf # CZPowGate
+    >>> assert cirq.FSimGate(-np.pi / 4, phi) in gf # SQRT_ISWAP
+    >>> assert cirq.FSimGate(-np.pi / 8, phi) not in gf # No value of `phi` would make it equal to
+    >>>                                                 # any gate/gate type in `gates_to_accept`.
+    >>> assert cirq.CZ ** 0.25 not in gf # CZPowGate not in gate_types_to_check
+    >>> assert cirq.SQRT_ISWAP not in gf # ISwapPowGate not in gate_types_to_check
+    """
+
+    def __init__(
+        self,
+        *,
+        gates_to_accept: Sequence[Union[Type[POSSIBLE_FSIM_GATES], POSSIBLE_FSIM_GATES]] = (),
+        gate_types_to_check: Sequence[Type[POSSIBLE_FSIM_GATES]] = (),
+        allow_symbols: bool = False,
+        atol=_DEFAULT_ATOL,
+    ):
+        """Inits `cirq_google.FSimGateFamily`.
+
+        Args:
+            gates_to_accept: List of gate types or instances to be accepted. All elements
```

Comment:
I don't quite understand why this should include types as well as instances, when `gate_types_to_check` already contains gate types. Can you clarify?

Reply:
- `gate_types_to_check` puts an initial filter on which compatible gate instances should be checked for containment. 
- `gates_to_accept` defines the class of compatible gate instances that should be accepted. 

So, for example, if I want to accept all iswappow gates which can be specified as either an fsim gate or an iswap gate, I would use:
```python
>>> cg.FSimGateFamily(
    gates_to_accept=[cirq.ISwapPowGate], 
    gate_types_to_check=[cirq.FSimGate, cirq.ISwapPowGate]
)
>>> assert cirq.PhasedISwapPowGate(phase_exponent = 0, exponent = 0.5) not in gf
>>> assert cirq.ISWAP ** 0.5 in gf
```